modules/default/weather/forecast.njk

```diff
@@ -19,10 +19,16 @@
                 <td class="align-right min-temp">
                     {{ f.minTemperature | roundValue | unit("temperature") }}
                 </td>
-                {% if config.showPrecipitationAmount %}
-                    <td class="align-right bright precipitation">
-                        {{ f.precipitation | unit("precip") }}
-                    </td>
+                {% if (config.showPrecipitationAmount) and (f.precipitation > 0) %}
+                    {% if f.precipitationUnits %}	
+                        <td class="align-right bright precipitation">
+                            {{ f.precipitation }}{{ f.precipitationUnits }}
```

Comment:
Why not translate it like the else case does?                             {{ f.precipitation | f.precipitationUnits }}

Reply:
I will apologize as this is my first serious bit of JS development, so my templating skills are very basic. But in this instance there is really no translation needed since the Provider code (e.g. envcanada) can explicitly set the precipitation and units values it wants displayed. Based on this, I thought it would be cleaner/simpler to do it as I've done.  I did look at making no changes to forecast.njk and pushing this logic into the filtering provided by weather.js, but this seemed really messy for what it required. I'm open to suggestions!

Reply:
No worries about being fresh in JS development, we all were once :-) Anyway since I am not into weather stuff, what units can precipation have? 
If there are multiple units possible, I'd rather move that into the weather.js config, something like showPrecipitationAmountUnit (but only if they are not covered with existing units).

Reply:
I'm really trying to solve 2 issues:

1. Today, if the config units are metric, precipitation will be expressed as 'mm' for all forecast days. This does not really work for Canada in that we measure snow in 'cm' and rain in 'mm'. So... I need a way to override that for envcanada.
2. In Canada, it is possible to have snow on one day and rain on the next. So for envcanada I need a way to let each forecast day to have a unique precipitation unit if required.

Based on these 2 requirements, I came up with the solution to add a new element to weatherobject to hold a custom unit for precipitation. If that element gets populated in Provider code, then it will be displayed. If that element is not populated (i.e. left as null), then default processing/filtering for precipitation will take place. This lets envcanada have its custom units without breaking anything for the existing Providers.

For sure I could add a new config boolean - e.g showCustomPrecipitaionUnit or something like that (weather.js already supports showPrecipitationAmount to toggle precipitation presentation). The new boolean would basically replace my check for a non-null value in my new weatherobject element precipitationUnits. Maybe that is a safer method to determine whether or not display the custom vs. default precipitation units?

And thanks for for all the feedback. Who knew Canadian weather could be so much fun? ;)